students/merrix/session03/mymailroom.py

```diff
@@ -0,0 +1,101 @@
+import sys  # imports go at the top of the file
+from collections import defaultdict
+from operator import itemgetter
+
+donor_db = defaultdict(list)
+
+
+donor_db = {"William Gates":[653772, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen" : [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0]
+           }
+
+prompt = "\n".join(("Welcome to James Dot Donations!",
+          "Please choose from below options:",
+          "1 - Send a Thank you",
+          "2 - Create a report",
+          "3 - Send Letter",
+          "4 - To Exit",
+          ">>> "))
+
+
+def view_donor_db():
+    print (donor_db.keys())
+    choose = input("enter a name from the list or a new name to donate: ").title()
+    if choose not in donor_db.keys():
+        add_donor_db(choose)
+    elif choose in donor_db.keys():
+        donation = int(input("Enter an amount to donate to %s " % choose)) 
+        donor_db[choose].append(int(donation))
+        print (donor_db)
+        print ('''
+        Thank you %s for your donation of $%i to this organization. 
+        We really appreciate your generousity and hope to see you again at our upcoming events.
+        Sincerely Yours
+        James Dot
+        ''' % (choose, donation))
+
+
+
+
+def add_donor_db(choose):
+    new_donor = input("Do you want to add %s to the donation list? yes or no " % choose).title()
+    if new_donor == "Yes":
+        amount = input("Enter an amount to donate for %s " % choose)
+        donor_db.update({new_donor:amount})
+        print (donor_db)
+        print ('''
+            Thank you %s for your donation of $%i to this organizat1ion. We really appreciate your generousity and hope to see you again at our upcoming events.
+            Sincerely Yours
+            James Dot
+            ''' % (new_donor, int(amount)))
+
+
+
+
+
+def user_Report():
+    print('Donor Name  | ',  'Total Given | ', 'Num Gift  | ', 'Average Gift')
+    for name, donations in donor_db.items():
+        num_of_gift = len(donations)
+        total_given = int(sum(donations))
+        average_gift = int(total_given/num_of_gift)
+        print ('{0:<15}   {1:^10}     {2:^6}     {3:8}'.format(name, total_given, num_of_gift, average_gift))
+        
+
+def send_letters():
+    for donor_name in donor_db:
+        letter = open('{}.txt'.format(donor_name), "w+")
```

Comment:
It's best to use `with open(...) as letter:`, so that the opened file is automatically closed. You are not using `with`, and not explicitly closing the written files, so you are risking losing your written content, if program crashes unexpectedly.